system-addon/lib/SnippetsFeed.jsm

```diff
@@ -47,16 +47,17 @@ this.SnippetsFeed = class SnippetsFeed {
       version: STARTPAGE_VERSION,
       profileCreatedWeeksAgo: profileInfo.createdWeeksAgo,
       profileResetWeeksAgo: profileInfo.resetWeeksAgo,
-
-      // TODO: There is currently no way to listen for changes to this value;
-      // bug is filed here https://bugzilla.mozilla.org/show_bug.cgi?id=1386318
       telemetryEnabled: Services.telemetry.canRecordBase,
-
       onboardingFinished: Services.prefs.getBoolPref(ONBOARDING_FINISHED_PREF)
     };
 
     this.store.dispatch(ac.BroadcastToContent({type: at.SNIPPETS_DATA, data}));
   }
+  _refreshCanRecordBase() {
+    // TODO: There is currently no way to listen for changes to this value;
+    // bug is filed here https://bugzilla.mozilla.org/show_bug.cgi?id=1386318
```

Comment:
Maybe mention that for now, we're doing the best effort of calling this each time a new tab is initialized.